blueocean-core-js/src/js/testutils.js

```diff
@@ -0,0 +1,49 @@
+import { Fetch, FetchFunctions } from './fetch';
+
+// default impls
+const fetchJSON = Fetch.fetchJSON;
+const fetch = Fetch.fetch;
+
+export default {
+    /**
+     * Switches fetch functions for ones that dont use JWT. Needed
+     * for running tests.
+     */
+    patchFetchNoJWT() {
+        Fetch.fetchJSON = FetchFunctions.rawFetchJSON;
+        Fetch.fetch = FetchFunctions.rawFetch;
+    },
+
+    /**
+     * Restores original fetch functions.
+     */
+    restoreFetch() {
+        Fetch.fetchJSON = fetchJSON;
+        Fetch.fetch = fetch;
+    },
+
+    /**
+     * Patches fetch functions with a resolved promise. This will make all fetch calls return
+     * this data.
+     *
+     * Usage
+     *
+     * TestUtils.patchFetchWithData((url, options) => {
+     *      assert.equals(url,"someurl")
+     *      return { mydata: 5 }
+     * })
+     */
+    patchFetchWithData(dataFn) {
```

Comment:
@kzantow there is at least one test thtat wants to verify the url, which is why i did it.